src/platform/libretro/memory.c

```diff
@@ -0,0 +1,38 @@
+/* Copyright (c) 2013-2014 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include "util/memory.h"
+
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/mman.h>
+
+#ifndef MAP_ANONYMOUS
+#define MAP_ANONYMOUS 0x20
+#endif
+
+#ifndef MAP_ANON
+#define MAP_ANON MAP_ANONYMOUS
+#endif
+#endif
+
+void* anonymousMemoryMap(size_t size) {
+#ifdef _WIN32
+	return VirtualAlloc(NULL, size, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
+#else
+	return mmap(0, size, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANON, -1, 0);
+#endif
+}
+
+void mappedMemoryFree(void* memory, size_t size) {
+#ifdef _WIN32
+	UNUSED(size);
+	// size is not useful here because we're freeing the memory, not decommitting it
+	VirtualFree(memory, 0, MEM_RELEASE);
+#else
+	munmap(memory, size);
+#endif
+}
```

Comment:
All of this code already exists in other places. I can vaguely understand the desire to have it all in one file and then have the preprocessor take care of it, but it's a needless duplication of code. At worst, you can use the preprocessor to include the individual files (e.g.

    #ifdef _WIN32
    #include "platform/windows/memory.c"
    #else
    #include "platform/posix/memory.c"
    #endif

...but there's a reason these directories exist in the first place. Since the goal is to support as many platforms as possible, you should probably just include each file in the Makefile as needed, as determined by the compilation target.